modules/stochastic_tools/test/tests/reporters/mapping/tests

```diff
@@ -0,0 +1,27 @@
+[Tests]
+  issues = '#23619'
+  design = 'MappingReporter.md'
+
+  [from_parallelstorage]
+    type = JSONDiff
+    requirement = 'The system should be able to map the solutions in a snapshot container into a latent space.'
+    input = map_main.i
+    jsondiff = 'map_training_data.json'
+    min_parallel=2
+    max_parallel=2
+    petsc_version = ">=3.8"
+  []
+  [from_solutions]
+    # Note: this test should result in the same reporter values as the one above. Due to reporter naming difference, we
+    # needed two different gold files. If something causes mismatch between the values in the two gold files, look for bugs.
```

Comment:
Well, we need to rename the reporter keys too. One of them is stored in a `data:value` format, while the other one is set manually on them main app. The reporter can be used in two very different ways and based on that we have different names:
1. You can use it on the subapp and then transfer the results to the main app for collection
2. You can call it in the main app on already collected results 
I would not like to alter Schemadiff for this. We are talking about 100 lines in the source code. But I agree, this can be dangerous if somebody alters one of them without looking at the note.